src/pyodide/internal/snapshot.ts

```diff
@@ -302,7 +334,17 @@ export function preloadDynamicLibs(Module: Module): void {
       path = path.slice(dynlibPath.length);
       root = dynlibRoot;
     }
-    loadDynlibFromTarFs(Module, sitePackages, root, path.split('/'));
+
+    const pathSplit = path.split('/');
+    if (pathSplit[0] == '') {
+      // This is a file path beginning with `/`, like /session/metadata/vendor/pkg/lib.so. So we
+      // are loading the vendored package's dynlibs here.
+      loadDynlibFromVendor(Module, pathSplit, userBundleNames);
```

Comment:
Should we allow people to put `.so` files in their src directory too? If so, we can handle it in a followup if you want.

Reply:
Yeah, but let's do it later. I'll create a ticket.